src/Utility/HDF5Writer.C

```diff
@@ -359,8 +359,10 @@ void HDF5Writer::writeSIM (int level, const DataEntry& entry,
     else
       sim->extractPatchSolution(glbVec,pchVec,pch,ncmp,1);
 
-    field.resize(ncmp,pchVec.size()/ncmp);
-    field.fill(pchVec.ptr());
+    if (ncmp > 0) {
+      field.resize(ncmp,pchVec.size()/ncmp);
+      field.fill(pchVec.ptr());
+    }
```

Comment:
would be better with an early exit if (ncmp == 0) after L351 as the stuff in between will not anything useful in that case.

Reply:
ok